source/dub/dub.d

```diff
@@ -247,6 +247,75 @@ class Dub {
 		m_project = new Project(m_packageManager, pack);
 	}
 
+	/** Loads a single file package.
+
+		Single-file packages are D files that contain a package receipe comment
+		at their top. A recipe comment must be a nested `/+ ... +/` style
+		comment, containing the virtual recipe file name and a colon, followed by the 
+		recipe contents (what would normally be in dub.sdl/dub.json).
+
+		Example:
+		---
+		/+ dub.sdl:
+		   name "test"
+		   dependency "vibe-d" version="~>0.7.29"
+		+/
+		import vibe.http.server;
+
+		void main()
+		{
+			auto settings = new HTTPServerSettings;
+			settings.port = 8080;
+			listenHTTP(settings, &hello);
+		}
+
+		void hello(HTTPServerRequest req, HTTPServerResponse res)
+		{
+			res.writeBody("Hello, World!");
+		}
+		---
+
+		The script above can be invoked with "dub --single test.d".
+	*/
+	void loadSingleFilePackage(Path path)
+	{
+		import dub.recipe.io : parsePackageRecipe;
+		import std.file : mkdirRecurse, readText;
+
+		path = makeAbsolute(path);
+
+		string file_content = readText(path.toNativeString());
+		auto idx = file_content.indexOf("/+");
+		enforce(idx >= 0, "Missing /+ ... +/ recipe comment.");
+		file_content = file_content[idx+2 .. $];
+		
+		idx = file_content.indexOf("+/");
+		enforce(idx >= 0, "Missing closing \"+/\" for recipe comment.");
+		string recipe_content = file_content[0 .. idx];
+
+		idx = recipe_content.indexOf(':');
+		enforce(idx > 0, "Missing recipe file name (e.g. \"dub.sdl:\") in recipe comment");
+		auto recipe_filename = recipe_content[0 .. idx].strip();
+		recipe_content = recipe_content[idx+1 .. $];
+
+		auto recipe = parsePackageRecipe(recipe_content, recipe_filename);
+		recipe.buildSettings.sourceFiles[""] = [path.toNativeString()];
+		recipe.buildSettings.mainSourceFile = path.toNativeString();
+		if (recipe.buildSettings.targetType == TargetType.autodetect)
+			recipe.buildSettings.targetType = TargetType.executable;
+
+		auto root_path = m_tempPath ~ (".dub/single/"~recipe.name);
```

Comment:
It's unclear where root_path is used. I would have expected it to be passed to the Package constructor below. Or maybe it should be handled in a different place?